### PR TITLE
Add forwarded headers when proxying

### PR DESCRIPTION
**Pull request by giorgiosironi**
Also for https://github.com/microsimulation/ijm-xml/issues/60

~Possibly missing `X-Forwarded-Proto`?~

### CODE REVIEW

File: salt/microsimulation-demo/config/etc-nginx-sites-enabled-microsimulation-demo.conf [hunk 1]
```diff
@@ -6,6 +6,8 @@ server {
 
     location / {
         proxy_pass http://localhost:8080;
+        proxy_set_header X-Forwarded-Host dev--microsimulation-demo;
```
Comment (giorgiosironi) [inline salt/microsimulation-demo/config/etc-nginx-sites-enabled-microsimulation-demo.conf]: needs to be made dynamic, pillars probably

File: salt/microsimulation-demo/config/etc-nginx-sites-enabled-microsimulation-demo.conf [hunk 2]
```diff
@@ -6,6 +6,8 @@ server {
...
+        proxy_set_header X-Forwarded-Port 1280;
```
Comment (giorgiosironi) [inline salt/microsimulation-demo/config/etc-nginx-sites-enabled-microsimulation-demo.conf:16]: same here

Comment (giorgiosironi) [reply salt/microsimulation-demo/config/etc-nginx-sites-enabled-microsimulation-demo.conf:16]: add `X-Forwarded-Proto`?

**Review by thewilkybarkid — COMMENTED** [review]
https://github.com/microsimulation/demo/blob/67a5e9c34a4acd3135b4d2efde117d6653f3cdcd/journal/web/app.php#L21 expects `X-Forwarded-For` too.

**Comment by thewilkybarkid** [comment]
Is the proxying here only because the repo has `localhost` as the `server_name`? Could just have a Docker Compose file here that overrides `services.web.volumes` with real Nginx config?

**Comment by giorgiosironi** [comment]
> Is the proxying here only because the repo has localhost as the server_name?

The host-based nginx has HTTPS and standard setup like access logs that get to Loggly, so I'd avoid moving that into a container yet (consistent with what `annotations`, `profiles`, etc. do).

**Review by thewilkybarkid — APPROVED** [review]
Should add `X-Forwarded-For` too, but looks fine otherwise.

**Comment by giorgiosironi** [comment]
(forgot to push that)

**Comment by giorgiosironi** [comment]
No change, links are still in the form `http://localhost:8080/download/aHR0cDovL3dlYjo4MDgyLzAwMTEzL2lqbS0wMDExMy5wZGY=/ijm-00113.pdf?_hash=AKGWK0BfYIfQvgd496%2BaerNb3zlj72Zlr6Bn1mJ8mWU%3D`.

**Pull request merged**